diagnostics/surface_albedo_feedback/surface_albedo_feedback_calcs.py

```diff
@@ -0,0 +1,232 @@
+
+    for d in newdims[::-1]:
+        dims.insert(i, d)
+
+    for d in newdims:
+        _ = dims.pop(-1)
+
+    return A1.transpose(*dims)
+
+
+def climatology(field=None):
+    """Compute mean, std, trend, std of detrended, residuals
+
+    Parameters
+    ----------
+    field : xarray.DataArray, dims must be time, space (space can be multidim)
+    
+    Returns
+    -------
+    themean, thestd, trend, detrendedstd: xarray.DataArray, dims of month, space
+    residuals: xarray.DataArray, dims of year, month, space
+    """
+    
+    NY=int(field['time'].sizes['time']/12)
+
+    field=xr_reshape(field,'time',['year','month'],[np.arange(NY),np.arange(12)])
+    themean = field.mean(dim='year')
+
+    return themean
+
+def kernelalbedo(dt, ds, ut, us):
+
+    # solve for single layer reflections under Taylor '07 assumptions
+
+    alb = us/ds #surface albedo
+    absorb=(dt-ut-ds+us)/dt #atmospheric absorption (assumed first pass only)
+    R = (1-absorb-ds/dt)/(1-absorb-ds*alb/dt) #anayltitcally solution for single layer atmospheric reflection
+
+    kernel = (1./100.)*(dt*((1.-R)*(1.-R))*(1.-absorb)*(1.-alb*R+R))/((1.-R*alb)*(1.-R*alb)) #derivative of upwelling solar at TOA with respect to surface albedo --  in units of W m^-2 per .01 change in surface albedo
+    kernel.name = 'kernel'
+    
+    albedo = us/ds
+    albedo.name = 'albedo'
+
+    return kernel, albedo
+
+def globaltimemean(tas, fx):
+    t=tas.mean(dim='time')
+    glob=t*fx
+    glob=glob.sum()/fx.sum()
+    glob.name='Tglob'
+    
+    return glob
+
+def readandclimo(vname,file):
+
+    field = xr.open_dataset(file)
+    field = field[vname]
+    field = climatology(field)
+    
+    return field
+
+
+
+
+def process_data():
+
+    FSDT_var = "{FSDT_var}".format(**os.environ)
+    FSDS_var = "{FSDS_var}".format(**os.environ)
+    FSUT_var = "{FSUT_var}".format(**os.environ)
+    FSUS_var = "{FSUS_var}".format(**os.environ)
+    TAS_var  = "{TAS_var}".format(**os.environ)
+
+    wk_dir="{WK_DIR}".format(**os.environ)
+    
+    # these yrs only refer to the hist period for comparing kernel of mod to CERES
+    # this pod also uses piControl and 4XCO2 output, the years will differ with model
+    firstyr = "{FIRSTYR}".format(**os.environ)
+    lastyr = "{LASTYR}".format(**os.environ)
+
+    output_dir = wk_dir+'/model/'
+
+    ### model and obs data files and varnames: ###############################################
+
+    # procesed output file names
+    kernel_histmod_file=output_dir+'surface_albedo_kernel_'+firstyr+'-'+lastyr+'.nc'
+    kernel_pimod_file=output_dir+'surface_albedo_kernel_piControl.nc'
+    albedo_abmod_file=output_dir+'surface_albedo_abrupt-4xCO2.nc'
+
+
+    # process model piControl data to compute kerel, albedo, and Tglob
+    TAS_input_file = "{DATADIR}/mon/{CASENAME}.mon.{TAS_var}.piControl.nc".format(**os.environ)
+    area_input_file = "{DATADIR}/mon/areacella_fx_{model}_abrupt-4xCO2_r1i1p1f1_gn.nc".format(**os.environ)
```

Comment:
These definitions should be removed: absolute paths to all requested variables are set via the `path_variable` setting in the new settings file format, which defaults to `<variable name>_FILE`; see [documentation](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/ref_settings.html#varlist-entry-properties). Calls to open_dataset(), etc. below should also be changed.